src/order-history/OrderHistoryPage.jsx

```diff
@@ -16,27 +16,56 @@ class OrderHistoryPage extends React.Component {
     this.props.fetchOrders();
   }
 
-  renderOrder(order) {
-    return (
-      <div key={order.id}>ORDER ID: {order.id}</div>
-    );
+  getTableData() {
+    return this.props.orders.map(({
+      lineItems,
+      orderDate,
+      total,
+      currency,
+      orderId,
+      receiptUrl,
+    }) => ({
+      description: this.renderLineItems(lineItems),
+      order_date: <FormattedDate value={new Date(orderDate)} />,
+      // eslint-disable-next-line react/style-prop-object
+      total: <FormattedNumber value={total} style="currency" currency={currency} />,
+      receiptUrl: <Hyperlink destination={receiptUrl}>View Order Details</Hyperlink>,
+      orderId,
+    }), this);
   }
 
-  renderOrders() {
-    if (this.props.orders.length === 0) {
+  renderLineItems(lineItems) {
+    return lineItems.map(({
+      itemId,
+      description,
+      quantity,
+    }) => (
+      <p className="d-flex" key={itemId}>
+        <span className="mr-3">{quantity}x</span>
+        <span>{description}</span>
+      </p>
+    ));
+  }
+
+  renderOrdersTable() {
+    if (!this.props.loading && this.props.orders.length === 0) {
       return (
         <p>
           {this.props.intl.formatMessage(messages['ecommerce.order.history.no.orders'])}
         </p>
       );
     }
+
     return (
       <Table
-        data={this.props.orders}
+        className="order-history"
+        data={this.getTableData()}
```

Comment:
Huh - that `Table` requires us to give it a data structure with a bunch of JSX in it feels unfortunate... `getTableData` is going to be evaluated on each render.

Reply:
Yup :)